plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -115,15 +117,16 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
 
    // thread-safe
    std::optional<chain::block_id_type> get_block_id(uint32_t block_num) {
-      std::optional<chain::block_id_type> id;
       if( trace_log ) {
-         id = trace_log->get_block_id( block_num );
-         if( id )
+         if ( auto id = trace_log->get_block_id( block_num ); id )
```

Comment:
Small detail, you don't need the `id` here and following lines, so it could be:

```suggestion
         if ( auto id = trace_log->get_block_id( block_num ) )
```